pkg/gardenlet/controller/shoot/shoot_care_control.go

```diff
@@ -374,22 +376,16 @@ func (c *defaultCareControl) Care(shootObj *gardencorev1beta1.Shoot, key string)
 	}
 
 	// Mark Shoot as healthy/unhealthy
-	if _, err := kutil.TryUpdateShootLabels(
-		ctx,
-		gardenClient.GardenCore(),
-		retry.DefaultBackoff,
-		shoot.ObjectMeta,
-		shootpkg.StatusLabelTransform(
-			shootpkg.ComputeStatus(
-				shoot.Status.LastOperation,
-				shoot.Status.LastErrors,
-				conditionAPIServerAvailable,
-				conditionControlPlaneHealthy,
-				conditionEveryNodeReady,
-				conditionSystemComponentsHealthy,
-			),
-		),
-	); err != nil {
+	oldObj := shoot.DeepCopy()
+	kutil.SetMetaDataLabel(&shoot.ObjectMeta, common.ShootStatus, string(shootpkg.ComputeStatus(
+		shoot.Status.LastOperation,
+		shoot.Status.LastErrors,
+		conditionAPIServerAvailable,
+		conditionControlPlaneHealthy,
+		conditionEveryNodeReady,
+		conditionSystemComponentsHealthy,
+	)))
+	if err := gardenClient.DirectClient().Patch(ctx, shoot, client.MergeFrom(oldObj)); err != nil {
```

Comment:
This call doesn't read anything from the API server, so `Client` and `DirectClient` are behaving equally here.
```suggestion
	if err := gardenClient.Client().Patch(ctx, shoot, client.MergeFrom(oldObj)); err != nil {
```